src/production/module/ImportReference.ts

```diff
@@ -222,11 +239,14 @@ function exportSpecifierImport(props: IImportReferencesProps, scope: IImportRefe
   );
 }
 
-export function ImportReferences(productionContext: IProductionContext, module: Module) {
-  const references: Array<IImport> = [];
+export interface IImportReferences {
+  references: Array<IImport>;
+  register: (props: IImportReferencesProps) => void;
+}
 
+export function ImportReferences(productionContext: IProductionContext, module: Module): IImportReferences {
   const scope = {
-    references,
+    references: [],
```

Comment:
not sure if i need to default it to an empty array

Reply:
why not? xD